### PR TITLE
pkg emsdk

### PR DESCRIPTION
**Pull request by Ludea**
adding emsdk and emscripten

### CODE REVIEW

**Comment by Ludea** [comment]
is something wrong with build.sh ?

**Comment by Grimler91** [comment]
TERMUX_PKG_DESCRIPTION is a bit long and TERMUX_PKG_LICENSE is not set to a valid license, see the linter output: https://github.com/termux/termux-packages/pull/6037/checks?check_run_id=1403167992

We call the MIT/Expat license just MIT, so change to `TERMUX_PKG_LICENSE=MIT` and the linter should at least be happy and start the build

File: packages/emsdk/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+TERMUX_PKG_HOMEPAGE=https://emscripten.org/
+TERMUX_PKG_DESCRIPTION="Emscripten is a complete compiler toolchain to WebAssembly"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_VERSION=2.0.8
+TERMUX_PKG_SRCURL=https://github.com/emscripten-core/emsdk/archive/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=248d716dbe2e2424ee3f205c41024b4efde974e8987f2204a97edb315c588ff3
+TERMUX_PKG_DEPENDS="llvm"
+TERMUX_PKG_BUILD_DEPENDS="python3, node, java"
```
Comment (xeffyr) [inline packages/emsdk/build.sh:8]: Please note that build depends specify only Termux packages. Not packages of the host system. Also - one of specified package names available in Termux.

Don't forget that you cross-compiling package not only for different arch but for different platform.

File: packages/emsdk/build.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+termux_step_pre_configure() {
+	apt install -y python3
```
Comment (xeffyr) [inline packages/emsdk/build.sh:11]: Delete this, Docker image already has Python 3 installed: https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh#L57-L59

Your command is wrong as `./build-package.sh` is not executed as root. Sudo is not allowed in build steps either.

https://github.com/termux/termux-packages/wiki/Creating-new-package?

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/emsdk/build.sh (hunk 1 above)
Comment (xeffyr) [reply packages/emsdk/build.sh:8]:
```
ERROR: Package emsdk depends on non-existing package "python3"
ERROR: Obtaining buildorder failed
Error: Process completed with exit code 1.
```

File: packages/emsdk/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+TERMUX_PKG_HOMEPAGE=https://emscripten.org/
+TERMUX_PKG_DESCRIPTION="Emscripten is a complete compiler toolchain to WebAssembly"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_VERSION=2.0.8
+TERMUX_PKG_SRCURL=https://github.com/emscripten-core/emsdk/archive/${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=248d716dbe2e2424ee3f205c41024b4efde974e8987f2204a97edb315c588ff3
+TERMUX_PKG_DEPENDS="llvm"
+TERMUX_PKG_BUILD_DEPENDS="python nodejs"
```
Comment (xeffyr) [inline packages/emsdk/build.sh:8]: Wrong format. Packages should be delimited by commas.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/emsdk/build.sh (hunk 2 above)
Comment (Ludea) [reply packages/emsdk/build.sh]: Thanks for the link, I don't know about this wiki

**Comment by xeffyr** [comment]
Fails with
```
/home/builder/termux-packages/packages/emsdk/build.sh: line 11: ./emsdk: No such file or directory
```
Need `TERMUX_PKG_BUILD_IN_SRC=true`.

**Comment by Ludea** [comment]
I am travelling, I will try to build on local env with docker, with a computer, not on my phone ^^

**Pull request closed without merging**